### ISSUE TITLE
Interface for Anime item and Manga item

### ISSUE THREAD

**Issue by flashkid10** [jikan-me/jikan#209]
Is there a way to for you to add an Interface for the Anime item and Manga item so I can treat them as the same item to help with accessing internal property.

**Comment by flashkid10** [comment]
Something like this:
public interface MalInteracter
    {
        long MalId { get; }
        string Title { get; }
        string Studios { get; }
        string Type { get; }
        string Author { get; }
        string Genres { get; }
        float? Score { get; }
    }

**Comment by irfan-dahir** [comment]
I don't use interfaces for the models, they're factory classes. You can check if they're the instances of the Model returned.

```php
if ($obj instanceof \Jikan\Model\Manga\Manga or $obj instanceof \Jikan\Model\Anime\Anime) {
}
```

**Comment by janvernieuwe** [comment]
The PHP classes should have full annotations for autocompletion (I personally made sure of this) and these models are quite different.
Are you using this lib, or the api?

**Comment by flashkid10** [comment]
I'm using the library
I've created a class to contain the values I want from from Manga & Anime;
Instead of  
`
public long ID
        {
            get
            {
                if (AniItem != null) return AniItem.MalId;
                else return ManItem.MalId;
            }
        }
`

I want to reference the ID though a shared interface

**Comment by irfan-dahir** [comment]
@flashkid10 Are you possibly using the .NET REST API wrapper for Jikan?
https://github.com/Ervie/jikan.net

**Comment by flashkid10** [comment]
Yeah, Why? Is this the wrong place for the question?

**Comment by tybug** [comment]
Yep. Open this issue under https://github.com/Ervie/jikan.net/issues/new; if you're using the .NET wrapper, then that wrapper is the one you want to have the interface implemented in, not the PHP version.
